### PR TITLE
Migrate bgp addpath and gracefulrestart to new lib

### PR DESCRIPTION
**Pull request by DanG100**

### CODE REVIEW

**Review by dan-lepage — CHANGES_REQUESTED** [review]
LGTM, except for a few small comment fixes. Is it possible to have ygot generate interface types for the gostructs?  It would be nice if instead of having both `ConfigInterface` and `ConfigOCInterface` we could just have one function that handles both.

File: feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go [hunk 1]
```diff
@@ -216,26 +215,26 @@ func verifyOTGBGPTelemetry(t *testing.T, otg *otg.OTG, c gosnappi.Config, state
 	for _, d := range c.Devices().Items() {
 		for _, ip := range d.Bgp().Ipv4Interfaces().Items() {
 			for _, configPeer := range ip.Peers().Items() {
-				nbrPath := otg.Telemetry().BgpPeer(configPeer.Name())
-				_, ok := nbrPath.SessionState().Watch(t, time.Minute,
-					func(val *otgtelemetry.QualifiedE_BgpPeer_SessionState) bool {
-						return val.IsPresent() && val.Val(t).String() == state
-					}).Await(t)
+				nbrPath := gnmi.OTG().BgpPeer(configPeer.Name())
+				_, ok := gnmi.Watch(t, otg, nbrPath.SessionState().State(), time.Minute, func(val *ygnmi.Value[otgtelemetry.E_BgpPeer_SessionState]) bool {
```
[the remark this answers is not in the document]
Comment (DanG100) [reply feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go:219]: the otg yang is defining it's own enum, so the enums aren't (and shouldn't be) the same https://github.com/open-traffic-generator/models-yang/blob/main/models/bgp/open-traffic-generator-bgp.yang#L90 

if it imported to oc enum, then reusing the same type might be possible, (but probably difficult to implement and have caveats like what if a module augments the imported enum with a new type how would that work)

[the remark this answers is not in the document]
Comment (dan-lepage) [reply feature/bgp/addpath/otg_tests/route_propagation_test/route_propagation_test.go:219]: Yeah, I guess the real fix would involve standardizing a bunch of these enums and splitting them into a separate package, which would be a lot of work for very little actual payoff.

**Review by dan-lepage — APPROVED** [review]

File: internal/attrs/attrs.go [hunk 2]
```diff
@@ -57,7 +58,55 @@ func (a *Attributes) IPv6CIDR() string {
...
+		}
+		a6 := s6.GetOrCreateAddress(a.IPv6)
+		if a.IPv6Len > 0 {
+			a6.PrefixLength = ygot.Uint8(a.IPv6Len)
+		}
+	}
+	return intf
+}
+
+// ConfigInterface configures an OpenConfig interface with these attributes.
```
[the remark this answers is not in the document]
Comment (DanG100) [reply internal/attrs/attrs.go]: fixed comments. 
re generating a go interfaces, that'd be a lot of generated code for little benefit. Also for it to work all the methods would have to return interface types (the GetOrCreate...)

[the remark this answers is not in the document]
Comment (dan-lepage) [reply internal/attrs/attrs.go]: Oh, right. That's too bad - it means any helper that uses OC must be tied to a specific built library - but I don't think we can do anything about that.

**Pull request merged**